Adding terraform-module-standards.md page and changes to terraform-modules.md

dcallao opened the pull request:
This change introduces the Terraform Module Development Standards guide liked into the "Terraform Equinix Modules" page, under section 5. Develop your module, Follow the standards provided in the [Terraform Module Development Standards](https://file+.vscode-resource.vscode-cdn.net/Users/dcallao/Documents/workspace/equinix/equinix-labs/temp/equinix-labs/terraform-module-standards.md) guide.

displague commented on terraform-module-standards.md:
````diff
@@ -0,0 +1,340 @@
+### Naming Convention
+* All configuration objects should be named using underscores to delimit multiple words. This practice ensures consistency with the naming convention for resource types, data source types, and other prede ned values. Note that this convention does not apply to name [arguments](https://www.terraform.io/docs/configuration/resources.html#resource-types-and-arguments).
+
+```
````
The code snippets would benefit from "hcl" language being specified.

dcallao replied:
Thanks! This has been addressed on commit 029c82e089218dbf238fc2d3eaf5fe0b2930ce15

displague commented on terraform-module-standards.md:
```diff
@@ -0,0 +1,340 @@
+* Additional documentation should be stored in a `docs/` subdirectory.
+
+### Naming Convention
+* All configuration objects should be named using underscores to delimit multiple words. This practice ensures consistency with the naming convention for resource types, data source types, and other prede ned values. Note that this convention does not apply to name [arguments](https://www.terraform.io/docs/configuration/resources.html#resource-types-and-arguments).
```
A markdown linter would insert a space between headings and bullet lists. (VS Code has one that will autocorrect these observations)

dcallao replied:
Thanks! This has been addressed on commit 029c82e089218dbf238fc2d3eaf5fe0b2930ce15

displague commented on terraform-module-standards.md:
````diff
@@ -0,0 +1,340 @@
+```
+provider "equinix" {
+  source = "equinix/equinix"
+  version = "~> 1.11.0-alpha.2"
````
If we are calling out `alpha` builds, we should call out the special behavior around these releases: https://developer.hashicorp.com/terraform/language/expressions/version-constraints#version-constraint-behavior

dcallao replied:
Thanks! This has been addressed on commit 029c82e089218dbf238fc2d3eaf5fe0b2930ce15

displague reviewed (commented):
Hashicorp changed the Terraform documentation site. I've noticed that many Google results are 404'ing now. The links in this doc may also be 404'ing.

If any of the sections of this document are directly copying external resources, we should cite those sources and verify a permissive license in the projects where that documentation is sourced (Terraform documentation has generally been made available in GitHub projects).

dcallao commented:
> Hashicorp changed the Terraform documentation site. I've noticed that many Google results are 404'ing now. The links in this doc may also be 404'ing.
> 
> If any of the sections of this document are directly copying external resources, we should cite those sources and verify a permissive license in the projects where that documentation is sourced (Terraform documentation has generally been made available in GitHub projects).

Thanks for the note. I have fixed the broken url links and also added text that cites "Portions of this page are modifications based on work created and [shared by Google](https://developers.google.com/readme/policies) and used according to terms described in the [Creative Commons 4.0 Attribution License](https://creativecommons.org/licenses/by/4.0/)."
